src/main/java/com/yelp/nrtsearch/server/luceneserver/highlights/Highlighter.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2023 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.highlights;
+
+import com.yelp.nrtsearch.server.luceneserver.field.TextBaseFieldDef;
+import java.io.IOException;
+import java.util.Map;
+import org.apache.lucene.index.IndexReader;
+
+public interface Highlighter {
+  String getName();
+
+  String[] getHighlights(
```

Comment:
Unless there is some harm in exposing the cache from SearchContext or benefit in isolating it to highlights, I don't see why we should create another context object just to store a map. I also think that by adding to SearchContext it will be available to other plugins and other use-cases as well.